### PR TITLE
0.5.0

### PR DESCRIPTION
**Pull request by Hoverbear**
**This comes after #181**.

After merge @hoverbear will publish.

Instead of waiting for #101 and some pending PRs, we'll roll those out in 0.6.0.

The reason for accelerating this release is that the `client-rust` CI fails on stable with Clippy due to our use of `feature()`. There may be other downstreams affected, so we're releasing this soon with this fix to help that problem.

### CODE REVIEW

**Comment by hicqu** [comment]
LGTM.

**Comment by Hoverbear** [comment]
I updated it to include joint consensus #101 . :) @hicqu and @brson PTAL again.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,43 @@
...
+- Corrected the `single_mem_node` example. (https://github.com/pingcap/raft-rs/pull/162)
+- Fixed typos. (https://github.com/pingcap/raft-rs/pull/159)
+- Adopted [Hashbrown](https://github.com/Amanieu/hashbrown) over `FxHash`. (https://github.com/pingcap/raft-rs/pull/160)
+- Corrected learner checking in `handle_transfer_leader`. (https://github.com/pingcap/raft-rs/pull/165)
+- Resolved some lints (https://github.com/pingcap/raft-rs/pull/174, https://github.com/pingcap/raft-rs/pull/168, https://github.com/pingcap/raft-rs/pull/142, https://github.com/pingcap/raft-rs/pull/124)
+- Fixed uses of `#[feature(_)]` so that we can build on stable cleanly. (https://github.com/pingcap/raft-rs/pull/180)
+
+# 0.4.0 - 2018-09-18
+
+- No longer scan the raft log when becoming a leader. (https://github.com/pingcap/raft-rs/pull/100)
```
Comment (zhangjinpeng1987) [inline CHANGELOG.md]: Duplicated with L43?

Comment (brson) [reply CHANGELOG.md]: Good spot. Yeah this needs to be deleted @Hoverbear

Comment (Hoverbear) [reply CHANGELOG.md]: Oh no 🙈

**Comment by Hoverbear** [comment]
Blocked on #181

**Comment by Hoverbear** [comment]
@brson PTAL, added joint consensus change.

**Pull request merged**